src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -881,5 +888,22 @@ public static CurrentProcessingFileInformation convertFromMapOfObjects(
                             .get(PERSISTED_CURRENT_PROCESSING_FILE_HASH)))
                     .build();
         }
+
+        public void convertNameToHash() {
```

Comment:
I don't think this is a good idea. Now callers have to remember to call this in order for it to work as intended. I liked how it was before given it has self contained inside the `CurrentProcessingFileInformation` now this has to be called outside of this class and if it is not, it breaks the internal functionality of this (breaking encapsulation) and opening the door for bugs 🐛

Reply:
Agree. I can put it back inside the original place, but move the order a little bit. Please check next commit.